_source/_posts/2023-01-23-keycloak-kubernetes-prod.md

```diff
@@ -0,0 +1,644 @@
+```
+**Note**: The name for public IP must be the gateway or monolith application base name, you can find it in the JDL. For the current example, the application base name is `gateway`.
+
+```
+application {
+  config {
+    baseName gateway
+    applicationType gateway
+```
+
+Find out the IP address with the following command:
+
+```shell
+gcloud compute addresses describe gateway-ip --global --format='value(address)'
+```
+
+Create a `kuberntes` folder at the root of the project, and run the generator:
+
+```shell
+jhipster kubernetes
+```
+
+Choose the following options when prompted:
+
+- Type of application: **Microservice application**
+- Root directory: **../**
+- Which applications? (select all)
+- Set up monitoring? **No**
+- Which applications with clustered databases? select **store**
+- Kubernetes namespace: **demo**
+- Docker repository name: **your-dockerhub-username**
+- Command to push Docker image: `docker push`
+- Enable Istio? **No**
+- Kubernetes service type? **Ingress**
+- Kubernetes ingress type? **Google Kubernetes Engine Ingress**
+- Root FQDN for ingress services: **\<public-ip\>.nip.io**
+- Use dynamic storage provisioning? **Yes**
+- Use a specific storage class? (leave empty)
+
+### Customize the configuration
+
+Let's Encrypt requires an email address to remind you to renew the certificate after 30 days before expiry. You will only receive this email if something goes wrong when renewing the certificate with cert-manager. Set the email address in the issuer yaml descriptor `kubernetes/cert-manager/letsencrypt-staging-issuer.yml`:
+
+```yml
+spec:
+  acme:
+    server: https://acme-staging-v02.api.letsencrypt.org/directory
+    email: <your-email>
+```
+
+### Deploy to Google Kubernetes Engine
+
+Create a Kubernetes cluster on Google Cloud:
+
+```shell
+gcloud container clusters create jhipster-cluster \
+--zone southamerica-east1-a \
+--machine-type n1-standard-4 \
+--enable-autorepair \
+--enable-autoupgrade
+```
+
+**Note**: you can choose a different zone and machine type.
+
+Then fetch the cluster credentials with:
+
+```shell
+gcloud container clusters get-credentials jhipster-cluster --zone southamerica-east1-a
+```
+
+`get-credentials` will update a `kubeconfig` file with appropriate credentials and endpoint information to point `kubectl` at a specific cluster in Google Kubernetes Engine.
+
+Install [cert-manager](https://cert-manager.io/docs/tutorials/getting-started-with-cert-manager-on-google-kubernetes-engine-using-lets-encrypt-for-ingress-ssl/) in your cluster:
+
+```shell
+kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.12.1/cert-manager.yaml
+```
+
+**Note**: If you see a `kubectl` error __The gcp auth plugin has been removed. __, see [https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke](https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke) for details on how to fix it.
```

Comment:
Should the "gcp auth plugin..." part be italicized? It shows up with the underscores in the rendered HTML.

Reply:
Fixed